src/website/shared/auth.py

```diff
@@ -0,0 +1,116 @@
+import logging
+from functools import lru_cache
+from typing import Any, cast
+
+from allauth.socialaccount.models import SocialAccount
+from django.conf import settings
+from django.contrib.auth.models import Group
+from github import Github
+from github.NamedUser import NamedUser
+from github.Organization import Organization
+from github.Team import Team
+
+from shared.models import NixMaintainer
+from shared.utils import get_gh
+
+github: Github = get_gh(per_page=100)  # 100 is the API limit
+logger = logging.getLogger(__name__)
+
+
+@lru_cache(maxsize=1)
+def get_gh_organization(orgname: str) -> Organization | None:
+    """
+    Return the Github Organization instance given an organization name.
+    """
+    try:
+        return github.get_organization(login=orgname)
+    except Exception:
+        logger.exception("Failed to get organization %s", orgname)
+        return None
```

Comment:
This is not needed as we're only caching two teams, so it's at most two calls to `get_organization` for an application lifetime.